File: tests/fixtures/php/t004_pass.php
<?php

class MathTest extends TestCase
{
    #[DataProvider('additionProvider')]
    public function test_add(int $a, int $b, int $expected): void
    {
        $this->assertEquals($expected, $a + $b);
    }

    public static function additionProvider(): array
    {
        return [
            [1, 2, 3],
            [4, 5, 9],
        ];
    }

    public function test_subtract(): void
    {
        $this->assertEquals(1, 3 - 2);
    }
}

File: tests/fixtures/rust/t004_pass.rs
use rstest::rstest;

#[rstest]
#[case(1, 2, 3)]
#[case(4, 5, 9)]
fn test_add(#[case] a: i32, #[case] b: i32, #[case] expected: i32) {
    assert_eq!(a + b, expected);
}
